homeassistant/components/fan/wemo.py

```diff
@@ -0,0 +1,322 @@
+"""
+Support for WeMo humidifier.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/fan.wemo/
+"""
+import asyncio
+import logging
+from datetime import timedelta
+import requests
+import async_timeout
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.fan import (
+    DOMAIN, SUPPORT_SET_SPEED, FanEntity,
+    SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.const import ATTR_ENTITY_ID
+
+DEPENDENCIES = ['wemo']
+SCAN_INTERVAL = timedelta(seconds=10)
+DATA_KEY = 'fan.wemo'
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_CURRENT_HUMIDITY = 'current_humidity'
+ATTR_TARGET_HUMIDITY = 'target_humidity'
+ATTR_FAN_MODE = 'fan_mode'
+ATTR_FILTER_LIFE = 'filter_life'
+ATTR_FILTER_EXPIRED = 'filter_expired'
+ATTR_WATER_LEVEL = 'water_level'
+
+SPEED_MINIMUM = 'minimum'
```

Comment:
Does that mean that I can't expose all 5 fan speeds from the humidifier without modifying the __init__.py for the fan domain? I was hoping to avoid doing that and just "extend" the speed options in my platform.

Reply:
Yes, only speeds available in the base component can be used. Changing the base component requires approval in an issue in our architecture repo.

https://developers.home-assistant.io/docs/en/entity_index.html#changing-the-entity-model

Reply:
That makes sense, and I can understand the arguments being made in that issue. I'll map the wemo speeds to the HASS speeds for now, until that is resolved.